### ISSUE TITLE
Styles in some custom:card configuration not working with templating in theme

### ISSUE THREAD

**Issue by MetroAffe** [thomasloven/lovelace-card-mod#361]
My Home Assistant version: 2024.2.5

My lovelace configuration method (GUI or yaml): GUI

What I am doing:
Templating in themes

What I expected to happen:
Red background :D

What happened instead:
Ignores the configuration which is made directly to the cards

**Minimal** steps to reproduce:

Theme:
```yaml
card-mod-card: |
  .type-tile {
    {% if is_state(config.entity, 'on') %}
      background-color: red !important
      {% endif %}
    }
```
Card (Not part of any hui-tile-card!):
```yaml
type: custom:mini-graph-card
entities:
  - sensor.schlafzimmer_temperature
card_mod:
  style: |
    ha-card {
      background-color: red !important;
    }
```
This happens for many other cards too like Atomic-Calender for example.
If I remove "config.entity" from templating lines everything works like expected:

```yaml
card-mod-card: |
  .type-tile {
    {% if 1 %}
      background-color: red !important
      {% endif %}
    }
```
Error messages from the browser console:
Nothing
---

**By putting an X in the boxes ([]) below, I indicate that I:**

- [X] Understand that this is a channel for reporting bugs, not a support forum (https://community.home-assistant.io/).

- [X] Have made sure I am using the latest version of the plugin.

- [X] Have followed the troubleshooting steps of the "Common Problems" section of https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins.

- [X] Understand that leaving one or more boxes unticked or failure to follow the template above may increase the time required to handle my bug-report, or cause it to be closed without further action.

**Comment by ildar170975** [comment]
https://github.com/thomasloven/lovelace-card-mod/issues/180

**Comment by MetroAffe** [comment]
Well, yes, this does the trick. And I found the error in the protocols now... It helps me a lot, thank you. I struggled for a few hours with it. :D

But I think this can stay open as a reminder to improve the code? Its an error who should be catched right?

**Comment by ildar170975** [comment]
Not sure if this needs an improvement. This is a whole user-side issue - using a variable which is not defined yet.

**Comment by MetroAffe** [comment]
I don’t know if this behavior is completely a user-issue because I added the {%templating code%} to a class (type-tile) where every card I used in my view has an entity defined. And for these cards, it was working perfectly with my first version. But somehow some other cards, maybe the custom ones without a working config, will also check this config and throw an error - even though the selector doesn’t select these other ones!
In other words:
If you will use templating anywhere, every card must have a config. Therefore, please ensure to add this information to the documentation, at least.

I hope you can understand what I mean.

**Comment by ildar170975** [comment]
> If you will use templating anywhere, every card must have a config.

Imho every card does have a config ))
This issue (config.entity is undefined) only happens when using in themes (at least in my experience).
Have no idea how themes are loaded; may be on some step this code is attempted to be executed when this variable is not defined yet.

**Comment by MetroAffe** [comment]
Yes and no.
In my opinion there are some cards where the entity is configured in some other way:

For examle:
```yaml
entities:
  - entity: calendar.**
  - entity: calendar.** 
  - entity: calendar.**
 ```

And I have some nested cards (mini-graph-card) these were also the ones who don’t work.

Thinking about that I‘m pretty sure thats the problem…

**Comment by ildar170975** [comment]
> In my opinion there are some cards where the entity is configured in some other way:

My words "every card does have a config" are valid.
I was not talking about a particular "config.entity" variable.
Surely this variable is not defined everywhere and not supposed to.
It is written in the Docs that you can use a "config" variable - it was not promised that "config.entity" is always & everywhrer available.

**Comment by MetroAffe** [comment]
Okay yes 👍🏻
So there are cards with „empty“ config..
But why is it needed to check the config of all cards if templating were used in themes for just one type? There is probably the case for improvement. But of course this is definitely not needed! My „problem“ is solved.
With some more configuration a workaround is possible. So as I said. I would recommend to add this Information to the documentation..

**Comment by ildar170975** [comment]
> So there are cards with „empty“ config..

It is not "empty config". The config.entity does not exist here, nothing else.
